src/System.Runtime.Serialization.Xml/tests/CanonicalizationTestHelper.cs

```diff
@@ -0,0 +1,210 @@
+        }
+
+        public static XmlReader CreateXmlReader(ReaderWriterType rwType, byte[] buffer, Encoding encoding)
+        {
+            return CreateXmlReader(rwType, buffer, encoding, XmlDictionaryReaderQuotas.Max);
+        }
+
+        public static XmlReader CreateXmlReader(ReaderWriterType rwType, Stream stream, Encoding encoding)
+        {
+            return CreateXmlReader(rwType, stream, encoding, XmlDictionaryReaderQuotas.Max);
+        }
+
+        public static XmlWriter CreateXmlWriter(ReaderWriterType rwType, Stream stream, Encoding encoding)
+        {
+            return CreateXmlWriter(rwType, stream, encoding, null);
+        }
+
+        public static XmlWriter CreateXmlWriter(ReaderWriterType rwType, Stream stream, Encoding encoding, IXmlDictionary dictionary)
+        {
+            XmlWriter result = null;
+            switch (rwType)
+            {
+                case ReaderWriterType.Binary:
+                    result = XmlDictionaryWriter.CreateBinaryWriter(stream, dictionary);
+                    break;
+                case ReaderWriterType.Text:
+                    result = XmlDictionaryWriter.CreateTextWriter(stream, encoding);
+                    break;
+                case ReaderWriterType.MTOM:
+                    result = XmlDictionaryWriter.CreateMtomWriter(stream, encoding, int.MaxValue, "myStartInfo", null, null, true, false);
+                    break;
+                case ReaderWriterType.WebData:
+                    XmlWriterSettings settings = new XmlWriterSettings();
+                    settings.Encoding = encoding;
+                    result = XmlWriter.Create(stream, settings);
+                    break;
+                case ReaderWriterType.WrappedWebData:
+                    XmlWriterSettings settings2 = new XmlWriterSettings();
+                    settings2.Encoding = encoding;
+                    result = XmlWriter.Create(stream, settings2);
+                    result = XmlDictionaryWriter.CreateDictionaryWriter(result);
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException("rwType");
+            }
+            return result;
+        }
+    }
+
+    public class Helper
+    {
+        public static void CompareArrays(byte[] array1, int offset1, byte[] array2, int offset2, int count)
+        {
+            for (int i = 0; i < count; i++)
+            {
+                if (array1[i + offset1] != array2[i + offset2])
+                {
+                    throw new Exception(String.Format("Error, arrays different at position {0} (array1[{1}]={2}, array2[{3}]={4})",
+                        i, i + offset1, array1[i + offset1], i + offset2, array2[i + offset2]));
+                }
+            }
+        }
+
+        public static void DumpToFile(string fileName, byte[] buffer, int offset, int count)
+        {
+            try
+            {
+                FileStream fs = File.Create(fileName);
```

Comment:
It seems the `fileName` passed to `DumpToFile` is just fileName. The file would be created at the working directory of the test. We may not have write permission to the working directory. Can you please change the method to write files to temp folder?